UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/POSAIOScenarioWithInvalidProductIdentifier.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Net;
+using FluentAssertions;
+using NUnit.Framework;
+using UKHO.PeriodicOutputService.API.FunctionalTests.Helpers;
+using static UKHO.PeriodicOutputService.API.FunctionalTests.Helpers.TestConfiguration;
+
+namespace UKHO.PeriodicOutputService.API.FunctionalTests.FunctionalTests
+{
+    [Category("POSAIOScenarioWithInvalidProductIdentifier")]
+    public class POSAIOScenarioWithInvalidProductIdentifier
+    {
+        private string fssJwtToken;
+        private static readonly POSWebJobApiConfiguration posWebJob = new TestConfiguration().POSWebJobConfig;
+        private static readonly POSFileDetails posDetails = new TestConfiguration().posFileDetails;
+
+        [OneTimeSetUp]
+        public async Task Setup()
+        {
+            AuthTokenProvider authTokenProvider = new();
+            fssJwtToken = await authTokenProvider.GetFssToken();
+
+            HttpResponseMessage apiResponse = MockHelper.ConfigureFMAio(posWebJob.MockApiBaseUrl, posWebJob.FMConfigurationInValidProductIdentifier);
+            apiResponse.StatusCode.Should().Be((HttpStatusCode)200);
+            await CommonHelper.RunWebJobAio();
+        }
+
+        [Test]
+        public async Task WhenICallFileDownloadEndpointWithInvalidAioProductIdentifier_ThenABatchWithErrorTxtIsGenerated()
+        {
+            HttpResponseMessage responseMessage = await FssBatchHelper.VerifyErrorTxtExist(fssJwtToken);
+            responseMessage.StatusCode.Should().Be((HttpStatusCode)200);
```

Comment:
so the 200 response is returned regardless, therefore put that up front as if its not a 200 then there is no need to go further.


Also where are all the other test scenario's on this endpoint. they can't all come back as a 200 surely

Reply:
Hi Ray, ProductIdentifier Endpoint is from the ESS, so all the scenarios you are looking can be found under ESS project.